### PR TITLE
Add documentation for implicit coercion in Hive connector

### PR DESCRIPTION
**Pull request by geraint0923**

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -370,6 +370,32 @@ Drop a schema::
...
+
+* Integral number (``TINYINT``, ``SMALLINT``, ``INTEGER``, and ``BIGINT``)
+  to ``VARCHAR(x)``
+
+* ``REAL`` to ``DOUBLE``
+
+* All the integral widening conversions, such as, ``TINYINT`` to ``SMALLINT``,
+  and ``SMALLINT`` to ``BIGINT``
+
+``NULL`` value is still coerced to be ``NULL``, and any coercion failure
```
Comment (electrum) [inline presto-docs/src/main/sphinx/connector/hive.rst]:
```
Any coercion failure will result in null, which is the same behavior
as Hive. For example, converting the string ``'foo'`` to a number,
or converting the string ``'1234'`` to a ``TINYINT`` (which has
a maximum value of ``127``).

**Comment by electrum** [comment]
See "outdated" comments

**Comment by geraint0923** [comment]
Addressed the comments.

**Comment by electrum** [comment]
screen shot 2016-11-10 at 10 33 42 am

**Comment by geraint0923** [comment]
Updated the doc according to the comment.

**Pull request merged**